### PR TITLE
[v5.2] Bump Buildah to v1.37.0

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
The final step of the SIC vendor dance.  This bumps the following as noted:

c/storage v1.55.0
c/image v5.32.0
c/common v0.60
c/buildah v1.37.0

This is in preparation of Podman v5.2

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@edsantiago advice please!  These two tests are failing:  
Failed tests (2):
 - 124 bud-preserve-subvolumes
 - 136 bud-volume-perms
 
and they were included a couple weeks ago in Buildah via this PR: https://github.com/containers/buildah/pull/5604.  In particular, this commit in the PR tweaked the two tests that are failing:   https://github.com/containers/buildah/pull/5604/commits/a7098c776a7fda585358c91496d35ce690ea9fb7

I thought they would work given the vendoring, but perhaps I'm missing a salient point?  I'll go follow the skip protocol, holler if that's the wrong route.

**Comment by TomSweeneyRedHat** [comment]
I've added a hold until @edsantiago can review and validate the test changes that I've made.

File: docs/source/markdown/options/compat-volumes.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+####>   podman build, farm build
+####> If file is edited, make sure the changes
+####> are applicable to all of those.
+#### **--compat-volumes**
+
+Handle directories marked using the VOLUME instruction (both in this build, and
+those inherited from base images) such that their contents can only be modified
+by ADD and COPY instructions. Any changes made in those locations by RUN
+instructions will be reverted. Before the introduction of this option, this
+behavior was the default, but it is now disabled by default.
```
Comment (Luap99) [inline docs/source/markdown/options/compat-volumes.md:11]: if the option does not work we cannot documented it

**Review by Luap99 — CHANGES_REQUESTED** [review]
Any new options must be wired properly into podman including remote api and swager docs.
For example https://github.com/containers/podman/commit/6b7f718b4b97779c0f0107dd19d5a7b4df13f3be

**Comment by Luap99** [comment]
Also overall I strongly suggest that we merge into main first then cherry-pick, doing two PRs at the same time is confusing and doubles the reviewers work.

File: docs/source/markdown/options/compat-volumes.md (hunk 1 above)
Comment (edsantiago) [reply docs/source/markdown/options/compat-volumes.md:11]: ...but if the option isn't documented we cannot merge (due to the man page crosscheck). Catch-22. As you said, the option needs to be wired in. (And also like you said, this needs to be done on main first)

**Comment by mheon** [comment]
Is this one good to merge now?

**Comment by edsantiago** [comment]
No, this is not ready to merge. (1) Shouldn't this be done on main first, per our guidelines and as @Luap99 mentioned above? (2) the new `compat-*` options have not been wired into podman. They are documented, but nonfunctional (tests are skipped), and that seems suboptimal from a UX POV.

**Comment by TomSweeneyRedHat** [comment]
Closing.  #23440 is the new attempt for the main branch.  If that goes, we do the vendor dance and then cherry-pick the second commit from there into this release-branch.

**Pull request closed without merging**